### PR TITLE
Make some lit test patterns more permissive for upstream InstCombine changes.

### PR DESCRIPTION
**Pull request by perlfu**
New InstCombine patterns have been added in upstream LLVM.
Fix lit tests to accommodate incoming changes by making them more permissive.
Specifically: https://reviews.llvm.org/D110247

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit de2d4ac0ae91c9c55f1858dbcf238b8cfc2aaf92
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev ad24e368a2e7a0dcebef9b18af68893051cda329</li>
<li>SPVGEN: dev 8705e87a8b4aaed55a89f79ed423612ca310bfdd</li>
<li>XGL: dev 7e2070be478a90852ca88c2aee36406b9d19c861</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 8207d5d4fbe8b5848ad54eee18977b5ab5aaa48e</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221132)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31197/55283 (56.4%)</li>
<li>Failed: 0/55283 (0.0%)</li>
<li>Not Supported: 24086/55283 (43.6%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31196/55283 (56.4%)</li>
<li>Failed: 0/55283 (0.0%)</li>
<li>Not Supported: 24087/55283 (43.6%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31540/55283 (57.1%)</li>
<li>Failed: 0/55283 (0.0%)</li>
<li>Not Supported: 23743/55283 (42.9%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34313/55283 (62.1%)</li>
<li>Failed: 0/55283 (0.0%)</li>
<li>Not Supported: 20970/55283 (37.9%)</li>
</ul></details>


</ul></details>

**Review by dstutt — APPROVED** [review]
LGTM - thanks

File: llpc/test/shaderdb/ObjUniformBlock_TestIndirectIndex_lit.frag [hunk 1]
```diff
@@ -25,10 +25,10 @@ void main()
 ; SHADERTEST-LABEL: {{^// LLPC}} pipeline patching results
 ; SHADERTEST: call <4 x i32> @llvm.amdgcn.s.buffer.load.v4i32(<4 x i32> %{{[0-9]*}}, i32 0, i32 0)
 ; SHADERTEST: call i32 @llvm.amdgcn.s.buffer.load.i32(<4 x i32> %{{[0-9]*}}, i32 0, i32 0)
-; SHADERTEST: call <4 x i32> @llvm.amdgcn.s.buffer.load.v4i32(<4 x i32> %{{[0-9]*}}, i32 %{{[0-9]*}}, i32 0)
+; SHADERTEST: call <4 x i32> @llvm.amdgcn.s.buffer.load.v4i32(<4 x i32> %{{[0-9]*}}, i32 %{{[0-9a-z.]*}}, i32 0)
```
Comment (jayfoad) [inline llpc/test/shaderdb/ObjUniformBlock_TestIndirectIndex_lit.frag:28]: Can you use `%{{\w+}}` everywhere instead? (I'm not actually sure if `\w` is supported but I think it should be.)

Comment (perlfu) [reply llpc/test/shaderdb/ObjUniformBlock_TestIndirectIndex_lit.frag:28]: I'll see if we cannot programmatically replace %{{[0-9]*}} in all the lit tests as a follow up PR.

**Pull request merged**